codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameReader.java

```diff
@@ -589,13 +589,20 @@ private void readUnknownFrame(ChannelHandlerContext ctx, ByteBuf payload, Http2F
     }
 
     /**
-     * If padding is present in the payload, reads the next byte as padding. Otherwise, returns zero.
+     * If padding is present in the payload, reads the next byte as padding. The padding also includes the one byte
+     * width of the pad length field. Otherwise, returns zero.
      */
-    private short readPadding(ByteBuf payload) {
+    private int readPadding(ByteBuf payload) {
         if (!flags.paddingPresent()) {
             return 0;
         }
-        return payload.readUnsignedByte();
+        return (short) (payload.readUnsignedByte() + 1);
+    }
+
+    private static int lengthWithoutPadding(int readableBytes, int padding) {
+        return padding == 0
+                ? readableBytes
+                : readableBytes - padding + 1;
```

Comment:
You might comment that the +1 is due to the pad length, which comes before most of the payload whereas the padding itself comes after.